templates/multistage-buildconfig.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata: 
+  name: jlinked-image
+spec:
+  source:
+    dockerfile: |
+	#Stage-1: ubi9-jlinked-image is builder image + application + jlinked JVM
+	# install additional system dependencies (for ubi-micro) to /mnt/jrootfs
+	FROM ubi9-jlinked-image AS ubi9-jlinked-image  # Replace this with appropriate base image
+	USER 0
+	RUN mkdir -p /mnt/jrootfs
+	RUN microdnf install --installroot /mnt/jrootfs --releasever 9 --setopt install_weak_deps=0 --nodocs -y \
+	    --config=/etc/dnf/dnf.conf \
+	    --noplugins \
+	    --setopt=cachedir=/var/cache \
+	    --setopt=reposdir=/etc/yum.repos.d \
+	    --setopt=varsdir=/etc/dnf/vars \
+	    grep gawk
+	RUN rm -rf /mnt/jrootfs/var/cache/* /mnt/jrootfs/var/lib/rpm /mnt/jrootfs/var/lib/dnf
+
+	#Stage-2:copy application JAR and jlinked JRE to runtime image
+	FROM registry.access.redhat.com/ubi9/ubi-micro AS lean-runtime
+	ARG JAVA_HOME=/usr/lib/jvm/java
+
+	COPY --from=ubi9-jlinked-image /mnt/jrootfs/ /
+	COPY --from=ubi9-jlinked-image /deployments /deployments
+	COPY --from=ubi9-jlinked-image /tmp/jre ${JAVA_HOME}
+	COPY --from=ubi9-jlinked-image /opt/jboss/container/ /opt/jboss/container/
+	# these are in the micro image
+	RUN rm -rf /var/lib/dnf /var/lib/rpm
+
+	ENV JAVA_HOME="${JAVA_HOME}" PATH="${JAVA_HOME}/bin:$PATH"
+	USER 185
+	CMD /opt/jboss/container/java/run/run-java.sh
+  strategy:
+    type: Docker
+  output:
+    to:
+      kind: ImageStreamTag
+      name: jlinked-image:latest
```

Comment:
The ImageStream referenced here is going to need to be created. We can do that in this YAML by prefixing the YAML for a basic empty ImageStream before the buildConfig - see this example in Josh's PR (from the beginning of that file up to and including the line `---`): https://github.com/jboss-container-images/openjdk/pull/454/files#diff-de1c200a3318053f0b8b70c6b2b82ad4c8cda9759276557292918dfa4f960f93R1